src/study_plans/services.py

```
@@ -273,6 +244,97 @@ def get_study_plan(self, plan_id: str, workspace_info: Dict = None) -> Optional[
+            result = self.collection.update_one(
+                {"_id": ObjectId(plan_id)},
+                {"$set": update_data}
+            )
+            
+            if result.modified_count == 0:
+                raise AppException("No se realizaron cambios", AppException.BAD_REQUEST)
+        except AppException:
+            # Re-lanzar excepciones AppException
+            raise
+        except Exception as e:
+            raise AppException(f"Error al actualizar plan: {str(e)}", AppException.BAD_REQUEST)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)